lms/djangoapps/ccx/tests/test_views.py

```diff
@@ -337,14 +353,15 @@ def test_edit_schedule(self, today):
             kwargs={'course_id': CCXLocator.from_course_locator(self.course.id, ccx.id)})
         response = self.client.get(url)
         schedule = json.loads(response.mako_context['schedule'])  # pylint: disable=no-member
+
         self.assertEqual(len(schedule), 2)
         self.assertEqual(schedule[0]['hidden'], False)
-        self.assertEqual(schedule[0]['start'], None)
-        self.assertEqual(schedule[0]['children'][0]['start'], None)
-        self.assertEqual(schedule[0]['due'], None)
-        self.assertEqual(schedule[0]['children'][0]['due'], None)
+        # If a coach does not override dates, then dates will be imported from master course.
+        self.assertEqual(schedule[0]['start'], str(self.chapters[0].start)[:-9])
+        self.assertEqual(schedule[0]['children'][0]['start'], str(self.sequentials[0].start)[:-9])
         self.assertEqual(
-            schedule[0]['children'][0]['children'][0]['due'], None
+            schedule[0]['children'][0]['due'],
+            str(self.sequentials[0].due)[:-9] if self.sequentials[0].due else None
```

Comment:
This seems like a lot of logic all in one line -- it's a bit hard to follow. Would you mind breaking it up a bit? I'd suggest:
```python
if sequentials[0].due:
    expected_due = str(self.sequentials[0].due)[:-9]
else:
    expected_due = None
self.assertEqual(schedule[0]['children'][0]['due'], expected_due)
```
Also, whats' the deal with the `[:-9]` string slice? I'm not sure why you're only asserting on part of the string, rather than the full string.